docs/getting-started/cash-app-pay.md

```diff
@@ -0,0 +1,220 @@
+| `Declined` | Customer has declined the Cash App Pay authorization and must start the flow over or choose a new payment method |
+| `PayKitExceptionState` | The general wrapper state for exceptions. These can range from integration errors to network errors. The exception states are emitted only for unrecoverable error states. |
+
+## Step 4: Implement Deep Linking Handling
+
+The authorization flow will bring Cash App to the foreground on the Customer’s device. After the Customer either authorizes or declines, your app must be returned to the foreground, which means we need a way to call your app from Cash App.  This is accomplished by [declaring an incoming intent][intent-filter]{:target="_blank"} filter on your app's Android Manifest and passing a corresponding redirect URI that uses the SDK when creating a customer request (as can be seen on the next step).
+
+Here’s an example how this integration looks like for your `AndroidManifest`:
+
+``` xml
+<intent-filter>
+  <action android:name="android.intent.action.VIEW" />
+
+  <category android:name="android.intent.category.DEFAULT" />
+  <category android:name="android.intent.category.BROWSABLE" />
+
+  <!-- Register the Cash PayKit redirect URI or URL. Change this accordingly in your app. -->
+  <data
+      android:scheme="cashpaykit"
+      android:host="checkout" />
+</intent-filter>
+```
+
+## Step 5: Create a Customer Request
+
+You can create a customer request as soon as you know the amount you’d like to charge or if you'd like to create an on-file payment request. We suggest that you create this request as soon as your checkout view controller loads, so that your Customer can authorize the request without delay.
+
+### Step 5A: Setup the Afterpay Cash App Pay Handler
+
+Setup a handler to manage the data when received from signing the order token. Alternatively this can be passed in as the `handler` parameter in step 5B.
+
+``` kotlin
+class ExampleActivity: Activity {
+  Afterpay.setCashAppHandler(object : AfterpayCashAppHandler {
+    override fun didReceiveCashAppData(cashAppData: CashAppSignOrderResult) {
+      when (cashAppData) {
+        is CashAppSignOrderResult.Success -> TODO("Create the PayKit customemr request")
+        is CashAppSignOrderResult.Failure -> TODO("Display an error and restart payment flow")
+      }
+    }
+  })
+}
+```
+
+### Step 5B: Sign the Order Token
+
+After retrieving the token from your server to server call, the order needs to be signed, so as to retrieve the JWT and associated data. This can be done either by the suspending function:
+
+``` kotlin
+Afterpay.signCashAppOrder(token)
+```
+
+or with the async version of it:
+
+``` kotlin
+Afterpay.signCashAppOrderAsync(token)
+```
+
+### Step 5C: Create a PayKit Customer Request
+
+To charge $5.00 as a one-time payment, your createrequest call might look like this (in the below example, `cashAppData` is the response object that is returned in the `didReceiveCashAppData` handler method):
```

Comment:
Does the amount need to be set in the server-to-server call? From from glancing through this, mentioning the amount made me expect to see it in the code. What about this?
```suggestion
To charge a one-time payment, your create request call might look like this (in the below example, `cashAppData` is the response object that is returned in the `didReceiveCashAppData` handler method, which would echo the amount from your server-to-server Create Checkout call):
```